restorer: undefined reference to atomic_dec_and_test

The following error seems related to https://github.com/checkpoint-restore/criu/commit/4a67a9a46ed32532915b31fcd6b6b8f066cdf94b
```
ld: warning: criu/pie/restorer.built-in.o has a LOAD segment with RWX permissions
ld: criu/pie/restorer.o: in function `thread_fini':
/root/criu/criu/pie/restorer.c:749: undefined reference to `atomic_dec_and_test'
ld: /root/criu/criu/pie/restorer.c:749: undefined reference to `atomic_dec_and_test'
make[2]: *** [/root/criu/scripts/nmk/scripts/build.mk:210: criu/pie/restorer.built-in.o] Error 1
make[1]: *** [criu/Makefile:59: pie] Error 2
make: *** [Makefile:278: criu] Error 2
```
https://github.com/checkpoint-restore/criu/actions/runs/22996260017/job/66773615298

--- a/include/common/arch/loongarch64/asm/atomic.h
+++ b/include/common/arch/loongarch64/asm/atomic.h
@@ -41,6 +41,8 @@ static inline int atomic_add_return(int i, atomic_t *v)
 #define atomic_inc_return(v)	atomic_add_return(1, v)
 #define atomic_dec(v)		atomic_sub(1, v)
 #define atomic_dec_return(v)	atomic_sub_return(1, v)
+/* true if the result is 0, or false for all other cases. */
+#define atomic_dec_and_test(v) (atomic_sub_return(1, v) == 0)
 
 static inline int atomic_cmpxchg(atomic_t *ptr, int old, int new)
 {

